sw/device/tests/dif/dif_alert_handler_unittest.cc

```diff
@@ -140,424 +140,426 @@ TEST_F(ConfigTest, BadClassPtr) {
-          .use_escalation_protocol = kDifAlertHandlerToggleDisabled,
-          .automatic_locking = kDifAlertHandlerToggleEnabled,
-          .accumulator_threshold = 12,
-          .irq_deadline_cycles = 30000,
-          .phase_signals = signals_a.data(),
-          .phase_signals_len = signals_a.size(),
-          .phase_durations = durations_a.data(),
-          .phase_durations_len = durations_a.size(),
-      },
-  };
-
-  dif_alert_handler_config_t config = {
-      .ping_timeout = 50,
-      .classes = classes.data(),
-      .classes_len = classes.size(),
-  };
-
-  classes[0].alerts = nullptr;
-  EXPECT_EQ(dif_alert_handler_configure(&handler_, config),
-            kDifAlertHandlerConfigError);
-  classes[0].alerts = alerts_a.data();
-
-  classes[0].local_alerts = nullptr;
-  EXPECT_EQ(dif_alert_handler_configure(&handler_, config),
-            kDifAlertHandlerConfigError);
-  classes[0].local_alerts = locals_a.data();
-
-  classes[0].phase_signals = nullptr;
-  EXPECT_EQ(dif_alert_handler_configure(&handler_, config),
-            kDifAlertHandlerConfigError);
-  classes[0].phase_signals = signals_a.data();
-
-  classes[0].phase_durations = nullptr;
-  EXPECT_EQ(dif_alert_handler_configure(&handler_, config),
-            kDifAlertHandlerConfigError);
-}
-
-TEST_F(ConfigTest, BadClass) {
-  IgnoreMmioCalls();
-
-  std::vector<dif_alert_handler_alert_t> alerts_a = {1, 2, 5};
-  std::vector<dif_alert_handler_local_alert_t> locals_a = {
-      kDifAlertHandlerLocalAlertEscalationPingFail,
-  };
-  std::vector<dif_alert_handler_class_phase_signal_t> signals_a = {
-      {.phase = kDifAlertHandlerClassStatePhase0, .signal = 3},
-      {.phase = kDifAlertHandlerClassStatePhase2, .signal = 1},
-  };
-  std::vector<dif_alert_handler_class_phase_duration_t> durations_a = {
-      {.phase = kDifAlertHandlerClassStateTerminal, .cycles = 20000},
-      {.phase = kDifAlertHandlerClassStatePhase2, .cycles = 15000},
-  };
-
-  std::vector<dif_alert_handler_class_config_t> classes = {
-      {
-          .alert_class = static_cast<dif_alert_handler_class_t>(12),
-          .alerts = alerts_a.data(),
-          .alerts_len = alerts_a.size(),
-          .local_alerts = locals_a.data(),
-          .local_alerts_len = locals_a.size(),
-          .use_escalation_protocol = kDifAlertHandlerToggleDisabled,
-          .automatic_locking = kDifAlertHandlerToggleEnabled,
-          .accumulator_threshold = 12,
-          .irq_deadline_cycles = 30000,
-          .phase_signals = signals_a.data(),
-          .phase_signals_len = signals_a.size(),
-          .phase_durations = durations_a.data(),
-          .phase_durations_len = durations_a.size(),
-      },
-  };
-
-  dif_alert_handler_config_t config = {
-      .ping_timeout = 50,
-      .classes = classes.data(),
-      .classes_len = classes.size(),
-  };
-
-  EXPECT_EQ(dif_alert_handler_configure(&handler_, config),
-            kDifAlertHandlerConfigError);
-}
+// TEST_F(ConfigTest, ClassInit) {
```

Comment:
@cfrantz can someone help look into this portion? It might be that the `difs` should be written slightly differently with this change also.

Reply:
@mcy  is probably the best person to look at the dif changes.

Reply:
I will look into fixing these as apart of fixing the DIF setting of the enable/class multireg now.